api/src/main/java/org/apache/iceberg/expressions/Literal.java

```diff
@@ -71,6 +71,18 @@ static Literal<BigDecimal> of(BigDecimal value) {
     return new Literals.DecimalLiteral(value);
   }
 
+  static Literal<Integer> ofDateLiteral(int value) {
```

Comment:
I don't think you should need any new literal constructors. You can always convert using `to` and the type.

Reply:
See my below comment.